Including Typing in requirements messes up python 3.7

In v0.35.x the addition of the typing library to the requirements conflicts with the builtin typing module on Python  3.7. 

I get the following error:
```
[ERROR] AttributeError: type object 'Callable' has no attribute '_abc_registry'
Traceback (most recent call last):
  File "/var/lang/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/var/lang/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/var/task/extraction/load_extractor_config.py", line 4, in <module>
    from typing import List
  File "/var/task/typing.py", line 1373, in <module>
    class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
  File "/var/task/typing.py", line 1024, in __new__
    self._abc_registry = extra._abc_registry
```

--- a/pypika/__init__.py
+++ b/pypika/__init__.py
@@ -85,4 +85,4 @@ from pypika.utils import (
 
 __author__ = 'Timothy Heys'
 __email__ = 'theys@kayak.com'
-__version__ = '0.35.3'
+__version__ = '0.35.4'

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -3,4 +3,3 @@ Sphinx==1.6.5
 sphinx-rtd-theme==0.2.4
 bumpversion==0.5.3
 parameterized==0.7.0
-typing==3.6.2

--- a/setup.py
+++ b/setup.py
@@ -51,9 +51,7 @@ setup(
       # Include additional files into the package
       include_package_data=True,
 
-      install_requires=[
-          'typing==3.6.2'
-      ],
+      install_requires=[],
 
       # Details
       url="https://github.com/kayak/pypika",

